cli/popper/commands/cmd_add.py

```diff
@@ -47,16 +45,35 @@ def cli(ctx, pipeline, folder, branch):
 
     owner, repo, pipe_name = pipeline.split('/')
 
+    if folder:
+        path, basename = os.path.split(folder)
+        if not basename:  # Only true when trailing slash is present
+            path, basename = os.path.split(path)
+        if basename:
+            new_pipe_name = basename  # Use basename as the name of pipeline
+        else:
+            pu.fail("The path is invalid.")
+    else:  # If no pipeline name is provided, use the original pipeline name
+        new_pipe_name = pipe_name
```

Comment:
can we consolidate this and the logic in https://github.com/systemslab/popper/blob/master/cli/popper/commands/cmd_init.py#L93 so that we create a `utils.get_name_and_path_for_new_pipeline()` function that takes as input the name of the pipeline and returns a two-element tuple `(name, path)`.